### PR TITLE
Add support for default cipher suites and `accepted-ciphers` config in C++

### PR DESCRIPTION
**Pull request by vekterli**
@havardpe @bjorncs please review

Since the TLS config file uses IANA cipher names but OpenSSL uses
its own cipher spec format internally, we explicitly remap the
provided names. We only support a modern subset of ciphers.

The default cipher suite contains ciphers that work across both
TLSv1.2 and TLSv1.3.

Currently no way to configure non-AEAD ciphers (AES CBC etc); we
consider this a feature rather than a bug.

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

**Review by havardpe — APPROVED** [review]

File: vespalib/src/vespa/vespalib/net/tls/transport_security_options.h [hunk 1]
```diff
@@ -21,11 +22,19 @@ class TransportSecurityOptions {
         vespalib::string _cert_chain_pem;
         vespalib::string _private_key_pem;
         AuthorizedPeers  _authorized_peers;
+        std::vector<vespalib::string> _accepted_ciphers;
+
+        Builder();
```
[the remark this answers is not in the document]
Comment (vekterli) [reply vespalib/src/vespa/vespalib/net/tls/transport_security_options.h]: Will rename, it was originally an actual builder pattern that was repurposed.

**Comment by vekterli** [comment]
@havardpe PTAL

**Comment by vekterli** [comment]
Arg, conflicts. Will fix.

File: vespalib/src/vespa/vespalib/net/tls/impl/openssl_tls_context_impl.cpp [hunk 2]
```diff
@@ -469,4 +476,31 @@ void OpenSslTlsContextImpl::set_ssl_ctx_self_reference() {
     SSL_CTX_set_app_data(_ctx.get(), this);
 }
 
+void OpenSslTlsContextImpl::set_accepted_cipher_suites(const std::vector<vespalib::string>& ciphers) {
+    vespalib::string openssl_ciphers;
+    size_t bad_ciphers = 0;
+    for (auto& iana_cipher : ciphers) {
```
Comment (havardpe) [inline vespalib/src/vespa/vespalib/net/tls/impl/openssl_tls_context_impl.cpp]: there is a const deduction here as well...

**Review by bjorncs — APPROVED** [review]

**Review by havardpe — APPROVED** [review]

**Comment by vekterli** [comment]
Dear CLA and merge bots: please wake up and do your thing.

Much love,
-me

**Pull request merged**